### ISSUE TITLE
var-usages inside defmethod bodies have null from-var

### ISSUE THREAD

**Issue by volrath** [clj-kondo/clj-kondo#2776]
Hey, I found this quirky behavior while using the analyzer to build an internal tool and I had Claude give it a shot at fixing it. Let me know if you think this is useful, otherwise happy to close both the issue and the PR. Here's Claude's description:

## Problem

When analyzing `defmethod` bodies, `var-usages` in the analysis output have `from-var` set to `null`. Calls inside `defn` bodies correctly get `from-var` set to the enclosing function name, but `defmethod` doesn't establish this context.

## Reproduction

```clojure
(ns test.defmethod
  (:require [clojure.string :as str]))

(defn helper [x]
  (str/upper-case x))

(defmulti greet :type)

(defmethod greet :formal
  [{:keys [name]}]
  (str/join " " ["Dear" (helper name)]))

(defmethod greet :casual
  [{:keys [name]}]
  (str/lower-case (str "hey " name)))
```

```bash
clj-kondo --lint /tmp/test_defmethod.clj \
  --config '{:output {:format :json} :analysis {:arglists true}}' \
  2>/dev/null | jq '.analysis["var-usages"][] | {row, name, from, "from-var", to}'
```

### Expected

Calls inside defmethod bodies should have `from-var` set to the multimethod name:

```json
{"row":11,"name":"join","from":"test.defmethod","from-var":"greet","to":"clojure.string"}
{"row":11,"name":"helper","from":"test.defmethod","from-var":"greet","to":"test.defmethod"}
```

### Actual

`from-var` is `null` for all calls inside defmethod bodies:

```json
{"row":11,"name":"join","from":"test.defmethod","from-var":null,"to":"clojure.string"}
{"row":11,"name":"helper","from":"test.defmethod","from-var":null,"to":"test.defmethod"}
```

Meanwhile, calls inside `defn` bodies correctly have `from-var` set:

```json
{"row":5,"name":"upper-case","from":"test.defmethod","from-var":"helper","to":"clojure.string"}
```

## Root Cause

`analyze-defmethod` in `analyzer.clj` calls `analyze-fn` without setting `:in-def` on the context. In contrast, `analyze-def` (for `defn`) does `(assoc ctx :in-def var-name)` before analyzing the function body. Since `:in-def` is what gets mapped to `:from-var` in the analysis output, defmethod bodies produce `null`.

## Use Case

This matters for tools that build call graphs from clj-kondo analysis. Without `from-var`, calls inside defmethod bodies can't be attributed to any function, leaving gaps in dependency graphs and blast-radius analysis.

## PR

Fix submitted in #2777.

**Comment by borkdude** [comment]
A few reservations I have with this:

- `from-var` / in-def means: in which var _definition_ is this var used. a defmethod doesn't define a var, technically speaking.
- `from-var` is a simple name. the reason it became like this is that `from`, the top level namespace and `from-var` together constitute the fully qualified symbol of the var.  But defmethods can be used outside of the namespace where it is defined. So when combining from-var + from for these cases, leads to incorrect answers to the question: in which var def did this usage occur?

If it's valuable to tooling, we could add a new key perhaps.

Maybe clojure-lsp is also interesting in information about in which defmethods vars were used? cc @ericdallo
